aten/src/ATen/native/quantized/cpu/qnnpack/test/convolution.cc

```diff
@@ -1135,3 +1135,203 @@ TEST(CONVOLUTION_OP, 3x3d2x1_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(1, 1)
+      .kernelSize(3, 3)
+      .dilation(1, 2)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_3x3d2x1_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(1, 1)
+      .kernelSize(3, 3)
+      .dilation(2, 1)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_3x3d2x1_runtime_quant_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(1, 1)
+      .kernelSize(3, 3)
+      .dilation(2, 1)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8(ConvolutionOperatorTester::Mode::Runtime);
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5s2_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .subsampling(2)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5s1x2_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .subsampling(1, 2)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5s2x1_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .subsampling(2, 1)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5d2_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .dilation(2)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5d1x2_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .dilation(1, 2)
+      .groups(27)
+      .iterations(3)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5d2x1_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .dilation(2, 1)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8();
+}
+
+TEST(CONVOLUTION_OP, depthwise_5x5d2x1_runtime_quant_per_channel) {
+  ConvolutionOperatorTester()
+      .inputSize(15, 14)
+      .padding(2, 2)
+      .kernelSize(5, 5)
+      .dilation(2, 1)
+      .groups(27)
+      .iterations(3)
+      .per_channel(true)
+      .testQ8(ConvolutionOperatorTester::Mode::Runtime);
+}
```

Comment:
Should we test other input sizes too apart from `(15, 14)`? Same comment for groups

Reply:
Thanks for pointing it out. Original tests were similar in input size. I am not sure if it makes a big difference. I can try adding few other sizes but I try to have the same base covered that the original tests did.